packages/ui/src/fields/Select/Input.tsx

```diff
@@ -83,6 +83,9 @@ export const SelectInput: React.FC<SelectInputProps> = (props) => {
       label: matchingOption ? getTranslation(matchingOption.label, i18n) : value,
       value: matchingOption?.value ?? value,
     }
+    // If value is not present then render nothing, allowing select fields to reset to their initial 'Select an option' state
+  } else if (!value) {
```

Comment:
I think you should add a !(typeof value !== 'number') or some way to ensure that 0 can be set as a value.

Reply:
or could this just be the final else statement, is there really a need to do `else if (!value)`?